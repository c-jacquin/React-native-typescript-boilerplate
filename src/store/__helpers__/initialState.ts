import { AppState } from '../types'
import { initialState as languageState } from '../language/reducer'
// Import state here

const initialState: AppState = {
    language: languageState,
    // Insert state here
}

export default initialState
