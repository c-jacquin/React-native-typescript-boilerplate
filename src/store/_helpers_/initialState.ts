import { AppState } from '../types'
import { initialState as languageState } from '../language/reducer'
import { initialState as navigationState } from '../navigation/reducer'
// Import state here

const initialState: AppState = {
    language: languageState,
    navigation: navigationState,
    // Insert state here
}

export default initialState
