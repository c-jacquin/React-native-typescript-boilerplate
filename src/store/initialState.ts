import { AppState } from './types'
import { initialState as languageState } from './language/reducer'
import { initialState as navigationState } from './navigation/reducer'
import { initialState as pushNotificationState } from './pushNotification/reducer'
// Import state here

const initialState: AppState = {
    language: languageState,
    navigation: navigationState,
    pushNotification: pushNotificationState,
    // Insert state here
}

export default initialState
