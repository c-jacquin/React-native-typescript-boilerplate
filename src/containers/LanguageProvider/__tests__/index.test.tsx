import React from 'react'
import renderer from 'react-test-renderer'
import { LanguageProvider, mapStateToProps } from '../index'
import initialState from 'store/_helpers_/initialState'

describe('Language Provider', () => {
    it('should renders correctly', () => {
        const tree = renderer
            .create(
                <LanguageProvider messages={{}}>
                    <div>test</div>
                </LanguageProvider>
            )
            .toJSON()
        expect(tree).toMatchSnapshot()
    })

    describe('mapStateToProps function', () => {
        it('should return an object with the locale', () => {
            expect(
                mapStateToProps(initialState, { messages: [], children: {} })
            ).toEqual({
                locale: initialState.language.locale,
            })
        })
    })
})
