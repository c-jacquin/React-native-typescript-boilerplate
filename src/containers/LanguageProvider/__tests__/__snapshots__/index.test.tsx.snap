// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Language Provider should renders correctly 1`] = `
<div>
  test
</div>
`;
